### ISSUE TITLE
Clarification

### ISSUE THREAD

**Issue by AFDesignsLTD** [TransformerOptimus/SuperAGI#843]
I think there should be a feature to enable clarification (toggle-able) where maybe one of the constraints is that if the agent needs clarification, it can ask the user for answers it might have to achieve the goals and instructions. The UI should prompt the user for input for the agent continue its run.

**Comment by clappo143** [comment]
I like this idea. The ability for the user to intervene every now and then with just a few instructions to get the agent back on track would be super useful. Maybe the user should simply have the option to intervene at any stage if they want. Or, going by your suggestion, if the agent finds itself 'confused' (or stuck in a loop), it could prompt the user to provide further instructions - if the user doesn't respond after say 60s, the agent could just resume what it was doing. If this happens like three more times (i.e. confusion/loops and no guidance provided from user when requested), perhaps the agent pauses or ends the run? The likelihood of it just looping around munching on tokens until it hits max iterations is pretty high from that point in my experience..

**Comment by AFDesignsLTD** [comment]
Yeah I know its not a priority as the projects young and there is actual issues to deal with but I think should be on the cards as it could make agents way more effective.

**Comment by clappo143** [comment]
Have you experimented with 'Permission Mode' (default is 'God Mode') in the Advanced Settings when creating an agent? This basically does what we're talking about, but requires a human response every time the agent uses a tool, instead of only when it becomes stuck/confused, or if the user just wants to intervene once to help the agent with a particular issue. 

So it seems there is already some of this functionality built in to app. It could perhaps be reworked or used as the basis for the agent-human interaction system we are describing, where the user has the *option* (but is not required) to intervene with a message at any stage during the run, and the agent also has the option to request user guidance/clarification (but is not required to; and if it does ask for help but no response is received after n seconds, it carries on anyway - or pauses, or whatever.)

But I agree, not a priority atm and in the wider of scheme of things...And this feature is not exactly in the spirit of building 'autonomous' AI agents. But that said, it would definitely improve goal completion success rates and probably also help with development (e.g. with identifying the particularly troublesome tools/tasks that most often require human assistance to prevent recurrent errors, loops, irrelevant outputs etc)

**Comment by AFDesignsLTD** [comment]
I'd simply put an 'Interrupt Now' and 'Interrupt after Message' button on the active bot page.
